### DISCUSSION TITLE
Main determinants/factors of startup time?

### DISCUSSION THREAD

**Discussion by yingzhu146** [mfussenegger/nvim-dap#1211] [Q&A]
I use the debugger like an interactive repl for test driven development via neotest. I absolutely love this setup, but iteration cycles are a bit slowed down due to slow startup times when rerunning tests in pytest.

out of curiosity, what are the most likely culprits here? just pytest being slow, is this dap? is there anything about the config i can do that's faster (e.g. pass different args)?

running via this 

```lua
require("neotest").run.run({ strategy = "dap" })
require("neotest-python")({                    
  dap = {                                               
    justMyCode = false,                                 
    console = "integratedTerminal",                     
    stopOnEntry = false,
    subProcess = false,    
  },                                                    
  args = { "-vv", "-s" },                               
  runner = "pytest",                                    
  python = "/Users/yingzhu/mambaforge/bin/python",     
}),                                                     
```

vscode isn't much faster with identical launch.json so assume this isn't a nvim-dap question but figured most knowledgeable crowd here

**Comment by mfussenegger** [comment]
It's basically:

- Time to spawn debugpy
- And to intiailize the debug session
- And to spawn pytest

So it is a little slower than starting pytest in the commandline, but shouldn't be too bad. For me it's typically a couple hundred ms between test trigger and it stopping at a breakpoint with just nvim-dap-python.

Maybe neotest does some additional test discovery pass - not sure about that.
